esphome/components/http_request/ota/ota_http_request.cpp

```diff
@@ -256,28 +226,32 @@ bool OtaHttpRequestComponent::http_get_md5_() {
 
   ESP_LOGVV(TAG, "url_with_auth: %s", url_with_auth.c_str());
   ESP_LOGI(TAG, "Connecting to: %s", this->md5_url_.c_str());
-  this->http_init(url_with_auth);
-  if (!this->check_status()) {
-    this->http_end();
+  auto container = this->parent_->get(url_with_auth);
+  if (container == nullptr) {
+    ESP_LOGE(TAG, "Failed to connect to MD5 URL");
     return false;
   }
-  int length = this->body_length_;
-  if (length < 0) {
-    this->http_end();
+  size_t length = container->content_length;
+  if (length == 0) {
+    container->end();
     return false;
   }
   if (length < MD5_SIZE) {
-    ESP_LOGE(TAG, "MD5 file must be %u bytes; %u bytes reported by HTTP server. Aborting", MD5_SIZE,
-             this->body_length_);
-    this->http_end();
+    ESP_LOGE(TAG, "MD5 file must be %u bytes; %u bytes reported by HTTP server. Aborting", MD5_SIZE, length);
+    container->end();
     return false;
   }
 
-  this->bytes_read_ = 0;
   this->md5_expected_.resize(MD5_SIZE);
-  auto read_len = this->http_read((uint8_t *) this->md5_expected_.data(), MD5_SIZE);
-  this->http_end();
+  int read_len = 0;
+  while (container->get_bytes_read() < MD5_SIZE) {
+    read_len = container->read((uint8_t *) this->md5_expected_.data(), MD5_SIZE);
+    App.feed_wdt();
+    yield();
+  }
+  container->end();
 
+  ESP_LOGD(TAG, "Read len: %u, MD5 expected: %u", read_len, MD5_SIZE);
```

Comment:
This could probably be verbose
```suggestion
  ESP_LOGV(TAG, "Read len: %u, MD5 expected: %u", read_len, MD5_SIZE);
```